### PR TITLE
make get_owner_ids consistent with the user list of case sharing groups

### PR DESCRIPTION
**Pull request by czue**
@dannyroberts in my opinion this has just been wrong for ages.

`get_owner_ids` included groups that weren't case-sharing in the results. this is what was used in determining your sync footprint. 

as a result if you happened to have cases assigned to them (rare but definitely possible if you changed the case-sharing status) then they would immediately be purged from the phone (since they weren't in the group fixture) and then you would get 412s

(easiest read commit by commit)

### CODE REVIEW

**Comment by czue** [comment]
wait for tests

File: corehq/apps/locations/tests/test_location_groups.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        self.user.add_location_delegate(outlet2)
+        self.user.add_location_delegate(outlet3)
+        self.user.add_location_delegate(state)
+        self.user.save()
+        fixture = location_fixture_generator(self.user, '2.0')
+        self.assertEquals(len(fixture[0].findall('.//state')), 1)
+        self.assertEquals(len(fixture[0].findall('.//outlet')), 3)
+
+
+
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_location_groups.py]: W391 blank line at end of file

**Comment by dannyroberts** [comment]
Wow, nice find! I agree, this has been wrong forever.

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**